### PR TITLE
MC Land Detector: try to make simpler to tune and less likely to detect early takeoffs, plus improve logging

### PR DESCRIPTION
**Pull request by sfuhrer**
## Describe problem solved by this pull request
1) vertical speed threshold for land detection very low, plus complex to tune. The threshold is currently:
`min(0.9 * max (MPC_LAND_CRWL, 0.1) * 0.5, LNDMC_Z_VEL_MAX)`, which with the default values is 0.9x0.5x0.3m/s = 0.135m/s, so about 4 times lower than `LNDMC_Z_VEL_MAX and means that `LNDMC_Z_VEL_MAX` is not at all considered (breaking [the documentation of the land detector](https://docs.px4.io/main/en/advanced_config/land_detector.html#ground-contact) .
image

2) rotational movement (that is checked for land detection) not logged
3) pre-emptive "takeoff" detection during auto takeoff due to motor spool up triggering rotational movement to be above threshold
image

## Describe your solution
1) MC LandDetector: remove dependency on `MPC_LAND_SPEED` and `MPC_LAND_CRWL`
Don't consider these params for vertical speed threshold, and instead increase the default for `LNDMC_Z_VEL_MAX` and use solely that one. Makes the land detector outcome more predictable and its internal logic simpler, while the new default tuning is increasing the `vz` threshold from 0.135m/s to 0.25m/s. 
For setting the in_descend flag I added a constant (0.3m/s) instead of using the crawl speed for that.

2) add a `rotational_movement` to `land_detector_status` message.

3) Increase threshold for rotational and vertical movement not only during the first 2s after a landing, but always if `maybe_landed` is true (so during takeoff as well as after a landing was detected. That makes it harder to trigger a "not maybe landed" due to rotational or vertical movement during propeller spool up, as well as it keeps it consistent after landing, not just in the first 2s. Also makes the land detector logic a bit simpler and more predictable by removing this extra internal state. 

## Describe possible alternatives
- Instead of https://github.com/PX4/PX4-Autopilot/pull/20425/commits/7b6e2d4ae88abdaa47add4dcd45ebdfd7a7420e2 we could just keep https://github.com/PX4/PX4-Autopilot/pull/20425/commits/d687fcb2cd36446bb044884649f66c7fdc101599 (this keeps the 2s window after takeoff for higher thresholds, but I don't think it's needed).

## Test data / coverage
Minimal SITL testing. 

## Additional context
It used to be that the threshold widening was done after arming instead of after landing, but this was changed in https://github.com/PX4/PX4-Autopilot/pull/7772. Don't fully understand why. 
The `vz` threshold was changed (linked to crawl speed) in https://github.com/PX4/PX4-Autopilot/pull/19126, it used to be considerably higher before as not the crawl speed but the land speed setpoint was taken into account. 

In SITL I sometimes see a thrust spike during an auto takeoff just before the thrust ramp start. This then triggers a pre-emtptive takeoff detection, but I didn't get to the root cause yet. Not sure if reproducible on real vehicles. 
image

### CODE REVIEW

**Comment by MaEtUgR** [comment]
@sfuhrer I rebased on latest main (#20428) and added my suggestions. I'm not 100% sure there's no corner case and the thresholds are far enough apart with different configurations but that's a base we could test to have improvement before doing further simplification and improvement steps.

File: src/modules/land_detector/MulticopterLandDetector.cpp [hunk 1]
```diff
@@ -278,21 +263,21 @@ bool MulticopterLandDetector::_get_maybe_landed_state()
...
 	}
 
-	// Next look if all rotation angles are not moving.
-	const float max_rotation_scaled = math::radians(_param_lndmc_rot_max.get()) * landThresholdFactor;
-
-	matrix::Vector2f angular_velocity{_angular_velocity(0), _angular_velocity(1)};
-
-	if (angular_velocity.norm() > max_rotation_scaled) {
+	if (_angular_velocity.xy().norm() > max_rotation_threshold) {
+		_rotational_movement = true;
```
Comment (MaEtUgR) [inline src/modules/land_detector/MulticopterLandDetector.cpp:276]: There are early returns in this function. That's why the logged rotation_movement state very often doesn't update as expected. We need to change the structure.

**Comment by MaEtUgR** [comment]
I rebased this on main. Working on it.

**Review by MaEtUgR — APPROVED** [review]
@sfuhrer First of all thanks a lot for looking into this, saved me quite some time that you already made these useful suggestions!

In my eyes, this pr solves the issues you described and I've also seen multiple times before on other platforms. You did testing and it didn't break without serious misconfiguration and I did tests without any issues. So I suggest we merge this and follow up with further improvements. I have a draft to simplify the maybe landed logic and hence also solve the rotational movement not updating because of early returns but it needs separate testing and I suggest using a separate pr instead of just delaying this one.
